graphics.cpp

```diff
@@ -67,3 +46,35 @@ void Graphics::AddSprite(const Sprite& sprite, const SDL_Rect& destination,
   SDL_RenderCopy(renderer_, sprite.GetSpriteTexture(), &source, &destination);
   perf->StopTimer("render_copy");
 }
+
+void Graphics::AddSprite(AnimatedSprite& sprite, Perf* perf,
+                         std::string animation) {
+  if (sprite.GetVisible()) {
+    int frame = sprite.GetFrameIndex();
```

Comment:
This is odd to make "GetSourceRect" require the frame number to be passed in, when the object that knows the current frame number is the sprite itself. Can't GetSourceRect just take the animation only?